src/Qir/Runtime/stdlib/include/qir_stdlib.h

```diff
@@ -0,0 +1,230 @@
+
+    // Returns the length of the string, in terms of bytes.
+    // http://www.cplusplus.com/reference/string/string/size/
+    QIR_SHARED_API uint32_t __quantum__rt__string_get_length(QirString* str); // NOLINT
+
+    // Allocates space for a tuple requiring the given number of bytes and sets the reference count to 1.
+    QIR_SHARED_API QirTuple* __quantum__rt__tuple_create(int64_t); // NOLINT
+
+    // Adds the given integer value to the reference count for the tuple. Deallocates the tuple if the reference count
+    // becomes 0. The behavior is undefined if the reference count becomes negative.
+    QIR_SHARED_API void __quantum__rt__tuple_update_reference_count(QirTuple*, int32_t); // NOLINT
+
+    // Adds the given integer value to the alias count for the tuple. Fails if the count becomes negative.
+    QIR_SHARED_API void __quantum__rt__tuple_update_alias_count(QirTuple*, int32_t); // NOLINT
+
+    // Creates a shallow copy of the tuple if the user count is larger than 0 or the second argument is `true`.
+    QIR_SHARED_API QirTuple* __quantum__rt__tuple_copy(QirTuple*, bool force); // NOLINT
+
+    // Returns a string representation of the big integer.
+    QIR_SHARED_API QirString* __quantum__rt__bigint_to_string(QirBigInt*); // NOLINT
+
+    // Creates a big integer with the specified initial value.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_create_i64(int64_t); // NOLINT
+
+    // Creates a big integer with the initial value specified by the i8 array. The 0-th element of the array is the
+    // highest-order byte, followed by the first element, etc.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_create_array(int, char*); // NOLINT
+
+    // Adds the given integer value to the reference count for the big integer. Deallocates the big integer if the
+    // reference count becomes 0. The behavior is undefined if the reference count becomes negative.
+    QIR_SHARED_API void __quantum__rt__bigint_update_reference_count(QirBigInt*, int32_t); // NOLINT
+
+    // Returns the negative of the big integer.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_negate(QirBigInt*); // NOLINT
+
+    // Adds two big integers and returns their sum.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_add(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Subtracts the second big integer from the first and returns their difference.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_subtract(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Multiplies two big integers and returns their product.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_multiply(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Divides the first big integer by the second and returns their quotient.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_divide(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Returns the first big integer modulo the second.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_modulus(QirBigInt*, QirBigInt*); // NOLINT
```

Comment:
FYI: I filed a QIR Spec issue [__quantum__rt__bigint_divide: The division by zero is not documented](https://github.com/qir-alliance/qir-spec/issues/27)